# Video 10-12: Downloading ChEMBL Data for COX-2
Explanation Script:
"Let's download all bioactivity data for human COX-2. We'll fetch compounds tested against this target, including their structures and IC50 values. After downloading, we'll save the data to a CSV file so we don't have to re-download it every time we work on this project."

In [1]:
from chembl_webresource_client.new_client import new_client
import pandas as pd
from IPython.display import display

# Get activities for COX-2 (CHEMBL230)
activities = new_client.activity
target_chembl_id = 'CHEMBL230'

print(f"Fetching bioactivity data for target {target_chembl_id}...")
res = activities.filter(target_chembl_id=target_chembl_id).filter(standard_type="IC50")

# Convert to DataFrame
df = pd.DataFrame(res)
print(f"Retrieved {len(df)} bioactivity records")

# Check available columns
print("\nAvailable columns in the data:")
print(df.columns.tolist())
print(f"\nTotal columns: {len(df.columns)}")

# Select columns that actually exist
available_cols = df.columns.tolist()
relevant_cols = ['molecule_chembl_id', 'standard_value', 'standard_units', 
                 'assay_description', 'document_year', 'canonical_smiles']

# Only keep columns that are in the dataframe
existing_cols = [col for col in relevant_cols if col in available_cols]
print(f"\nSelecting these available columns: {existing_cols}")

df_subset = df[existing_cols].copy()
print("\nSample records (first 10 rows):")
display(df_subset.head(10))

print(f"\nTotal records: {len(df)}")
print(f"Unique molecules: {df['molecule_chembl_id'].nunique()}")
if 'document_year' in df.columns:
    print(f"Date range: {df['document_year'].min()} - {df['document_year'].max()}")

# Show data summary
print("\n" + "=" * 60)
print("DATA SUMMARY")
print("=" * 60)
summary_df = pd.DataFrame({
    'Metric': ['Total Records', 'Unique Molecules', 'Date Range', 'Columns'],
    'Value': [
        f"{len(df)}", 
        f"{df['molecule_chembl_id'].nunique()}",
        f"{df['document_year'].min()} - {df['document_year'].max()}" if 'document_year' in df.columns else 'N/A',
        f"{len(df.columns)}"
    ]
})
display(summary_df)

# Save the full raw data to CSV
df.to_csv('cox2_chembl_raw_data.csv', index=False)
print(f"\n✓ Data saved to 'cox2_chembl_raw_data.csv'")
print(f"✓ File contains {len(df)} records with {len(df.columns)} columns")

C:\Users\HP\anaconda3\Lib\site-packages\chembl_webresource_client\__init__.py:4: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  __version__ = __import__('pkg_resources').get_distribution('chembl_webresource_client').version


Fetching bioactivity data for target CHEMBL230...
Retrieved 8383 bioactivity records

Available columns in the data:
['action_type', 'activity_comment', 'activity_id', 'activity_properties', 'assay_chembl_id', 'assay_description', 'assay_type', 'assay_variant_accession', 'assay_variant_mutation', 'bao_endpoint', 'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment', 'data_validity_description', 'document_chembl_id', 'document_journal', 'document_year', 'ligand_efficiency', 'molecule_chembl_id', 'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value', 'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id', 'standard_flag', 'standard_relation', 'standard_text_value', 'standard_type', 'standard_units', 'standard_upper_value', 'standard_value', 'target_chembl_id', 'target_organism', 'target_pref_name', 'target_tax_id', 'text_value', 'toid', 'type', 'units', 'uo_units', 'upper_value', 'value']

Total columns: 46

Selecting these available columns:

,molecule_chembl_id,standard_value,standard_units,assay_description,document_year,canonical_smiles
0,CHEMBL297008,60.0,nM,In vitro inhibitory activity against human pro...,1997.0,Cc1ccc(-c2ccc(S(C)(=O)=O)cc2)n1-c1ccccc1
1,CHEMBL289813,3230.0,nM,In vitro inhibitory activity against human pro...,1997.0,Cc1c(C=O)cc(-c2ccc(S(C)(=O)=O)cc2)n1-c1ccc(F)cc1
2,CHEMBL43736,80.0,nM,In vitro inhibitory activity against human pro...,1997.0,Cc1c(COc2cccc(Cl)c2)cc(-c2ccc(S(C)(=O)=O)cc2)n...
3,CHEMBL140167,0.12,nM,Tested in vitro for inhibition against Prostag...,2000.0,Fc1ccc(-c2[nH]c(-c3ccc(F)cc3)c3c2C2CCC3CC2)cc1
4,CHEMBL44194,100000.0,nM,In vitro inhibitory activity against human pro...,1997.0,CCc1ccc(-c2ccc(S(C)(=O)=O)cc2)n1-c1ccc(F)cc1
5,CHEMBL44290,1610.0,nM,In vitro inhibitory activity against human pro...,1997.0,CC(=O)c1cc(-c2ccc(S(C)(=O)=O)cc2)n(-c2ccc(F)cc...
6,CHEMBL44464,None,nM,In vitro inhibitory activity against human Pro...,1997.0,COC(=O)c1ccc(-n2c(C)ccc2-c2ccc(F)cc2)cc1
7,CHEMBL288773,100000.0,nM,In vitro inhibitory activity against human pro...,1997.0,Cc1ccc(-c2ccc(S(C)(=O)=O)cc2)n1Cc1ccc(F)cc1
8,CHEMBL44068,100000.0,nM,In vitro inhibitory activity against human pro...,1997.0,Cc1ccc(-c2ccc(S(C)(=O)=O)cc2)[nH]1
9,CHEMBL44367,100000.0,nM,In vitro inhibitory activity against human pro...,1997.0,CS(=O)(=O)c1ccc(-n2cccc2-c2ccc(F)cc2)cc1Cl



Total records: 8383
Unique molecules: 6349
Date range: 1994.0 - 2024.0

DATA SUMMARY


,Metric,Value
0,Total Records,8383
1,Unique Molecules,6349
2,Date Range,1994.0 - 2024.0
3,Columns,46



✓ Data saved to 'cox2_chembl_raw_data.csv'
✓ File contains 8383 records with 46 columns


Explanation of Output:
"We successfully retrieved 8,383 bioactivity records for COX-2 from 6,349 unique molecules. The data spans from 1994 to 2024 and includes SMILES structures. The complete dataset is now saved as 'cox2_chembl_raw_data.csv' in your working directory for future use without re-downloading."